### DISCUSSION TITLE
How to debug my python file with Hydra in Vscode launch.json？

### DISCUSSION THREAD

**Discussion by ghost** [facebookresearch/hydra#2334] [Q&A] [answered]
I'm using Hydra which requires passing additional arguments using a plus and then an equal sign between the argument and its valueI, e.g.
`python evaluate.py '+model_path="logs/fc/version_1/checkpoints/epoch=1-step=2.ckpt"'`
I want to pass this in vscode to launch.json to the args field; however, I don't know how to do it properly because typically the argument and value parts are separated by a space and not an equal sign as for Hydra. So the following doesn't work:
`{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Python: Current File",
      "type": "python",
      "request": "launch",
      "program": "${file}",
      "console": "integratedTerminal",
      "args" : ["+model_path", "logs/fc/version_1/checkpoints/epoch=1-step=2.ckpt"]
    }
  ]
}`

**Comment by serhangul** [comment, marked as answer]
I had the same issue. Maybe not the most elegant solution but it works if you pass everything as a single argument:

`"args" : ["+model_path='logs/fc/version_7/checkpoints/epoch=19-step=3700.ckpt'"]`

Note that you need the single quotes to escape the remaining equal signs. For more information, please check Hydra's [override syntax](https://hydra.cc/docs/1.2/advanced/override_grammar/basic/).

Edit: From the Hydra docs on [Quoted Values](https://hydra.cc/docs/1.2/advanced/override_grammar/basic/#quoted-values):
> to include a single quote in a single quoted string, use \' (for double quotes in a double quoted string, use \")

**Comment by MD-Student** [comment]
I have tried this and it works in my case.
```json
"args": [
                "device=0",
                "semantic_inversion=true",
                "unconstrained=false",
                "spatial_perturbation=false",
                "temporal_perturbation=false",
                "num_steps=5000000",
                "warm_up.start_step=4000000",
                "warm_up.end_step=5000000",
                "keyframe_selection_scheme=uncond"
            ]
```
